### PR TITLE
Off Balance Disarms only if shoved into a wall or person.

### PR DESCRIPTION
**Pull request by keronshb**
## About The Pull Request

This changes Off Balance to only disarm if they were shoved into a wall or a person WHILE they're affected by Off Balance.

## Why It's Good For The Game

Letting the crew disarm anyone instantly with two shoves is too good for what it's worth.  It's like everyone has a free and better baton or shield bash.  This way they need to be knocked into something while off balance for them to lose grip.

## Changelog
:cl:
balance: Made it so Off Balance only disarms if they're shoved into a wall or person.
balance: Reduced Off Balance time to 2 seconds.
/:cl:

### CODE REVIEW

**Comment by The-Sun-In-Splendour** [comment]
Salt pr?

**Comment by silicons** [comment]
mmmmmm don't like this post, making shovies useless in melee without guns involved especially with sprint being removed soon is pretty awful lol.

**Comment by necromanceranne** [comment]
I made this change, and I actually was of two minds about it. I didn't want security to completely lose the potential of being able to disarm people after a baton hit, so maybe it'd be a good idea to change the off-balance effect to improve the standard disarm probability too. In addition, I felt the mechanic was so finicky (requiring two shoves normally without a baton) and difficult to pull off and so *deliberate* that anyone could see it coming long enough to stow a weapon before a second shove occurred that it wouldn't be too overpowering if we really don't want knockdowns disarming.

It makes sense for what security does for them to be able to get weapons out of peoples hands with their gear, but maybe this was a little too extreme and there are better ways to accomplish this.

Suffice to say, no, it wasn't unintentional, I just didn't clean up a comment. You can check the original pr that changed this effect to verify that. I forgot to update the body, but it's in the changelog, which is the player facing element and the bit that matters. https://github.com/Citadel-Station-13/Citadel-Station-13/pull/14443

**Comment by ghost** [comment]
> 
> 
> I made this change, and I actually was of two minds about it. I didn't want security to completely lose the potential of being able to disarm people after a baton hit, so maybe it'd be a good idea to change the off-balance effect to improve the standard disarm probability too. In addition, I felt the mechanic was so finicky (requiring two shoves normally without a baton) and difficult to pull off and so _deliberate_ that anyone could see it coming long enough to stow a weapon before a second shove occurred that it wouldn't be too overpowering if we really don't want knockdowns disarming.
> 
> It makes sense for what security does for them to be able to get weapons out of peoples hands with their gear, but maybe this was a little too extreme and there are better ways to accomplish this.
> 
> Suffice to say, no, it wasn't unintentional, I just didn't clean up a comment. You can check the original pr that changed this effect to verify that. I forgot to update the body, but it's in the changelog, which is the player facing element and the bit that matters. #14443

Shoving disarm is absolutely awful with melee weapons, because it just takes 2 consecutive pushes and then haha, there goes your weapon.

Besides, security already has the means to disarm consistently with the right click on the baton, followed with a regular disarm. In its current state this buffs the crew's power to absurdly broken levels with the amount of reliability they have in disarming someone with nothing but their hands. I can see it being acceptable for guns, since if you let a person get close that's on you, but melee is melee and not all weapons can be swapped/stashed or doing so would mean you make yourself a sitting duck just because you got lightly shoved once, to avoid being disarmed

**Comment by necromanceranne** [comment]
Yeah, I think in hindsight I'm not particularly impressed with the change and I wouldn't disagree with a revert.

**Comment by DeltaFire15** [comment]
> Besides, security already has the means to disarm consistently with the right click on the baton, followed with a regular disarm.

This is actually related to the same thing, the baton applies to same debuff as the shove does, so this nerfs the baton-related disarm aswell.


Anyways, an idea would be to replace the unconditional disarm if caused by that status with a "always disarm gun, disarm anything if you shove them into a solid object instead"

**Comment by ghost** [comment]
> 
> 
> > Besides, security already has the means to disarm consistently with the right click on the baton, followed with a regular disarm.
> 
> This is actually related to the same thing, the baton applies to same debuff as the shove does, so this nerfs the baton-related disarm aswell.
> 
> Anyways, an idea would be to replace the unconditional disarm if caused by that status with a "always disarm gun, disarm anything if you shove them into a solid object instead"

If that's the case then it'd be better to create a separate debuff that's applied by shoving specifically. But being pushed into walls to disarm any weapon is likewise not great. Especially since it encroaches upon shield bashing, which has a lengthy cooldown whether you hit or not. If it disarmed just ranged weapons then that'd be fine, but not melee

**Comment by silicons** [comment]
do what deltafire said tbh

**Comment by silicons** [comment]
and do NOT have it, if you do, that shoving someone who's down into a wall/person/table/whatever doesn't disarm (aka don't make it only disarm while they're up, meaning to disarm you'd have to shake them up)
that would be very silly and nonsensical.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -2046,11 +2046,15 @@ GLOBAL_LIST_EMPTY(roundstart_race_names)
...
-					"<span class='danger'>You drop \the [target_held_item]!!</span>", null, COMBAT_MESSAGE_RANGE)
-				append_message += ", causing them to drop [target_held_item]"
+		else if(target.has_status_effect(STATUS_EFFECT_OFF_BALANCE))
+			if(target_held_item)
+				if(shove_blocked)
+					if (target.buckled)
+						return
+					else if(target.dropItemToGround(target_held_item))
+						target.visible_message("<span class='danger'>[target.name] drops \the [target_held_item]!!</span>",
+							"<span class='danger'>You drop \the [target_held_item]!!</span>", null, COMBAT_MESSAGE_RANGE)
```
Comment (silicons) [inline code/modules/mob/living/carbon/human/species.dm:2056]: no extra !

File: code/__DEFINES/combat.dm [hunk 2]
```diff
@@ -158,7 +158,7 @@
 /// how long they're staggered for
 #define SHOVE_STAGGER_DURATION 35
 /// how long they're off balance for
-#define SHOVE_OFFBALANCE_DURATION 30
```
Comment (silicons) [inline code/__DEFINES/combat.dm:161]: @DeltaFire15 thoughts?
i'm against the change.

Comment (keronshb) [reply code/__DEFINES/combat.dm:161]: Two seconds feels a lot like 3-4.  The window is bigger than it lets on to be.  When it's at 3 seconds it feels like 4-5.  I could change it back to 3 if we really need it, but even allowing everyone to disarm anything by getting shoved into a wall/person is still really goofy.

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (keronshb) [reply code/modules/mob/living/carbon/human/species.dm:2056]: I'll fix this when the other change is ready too.

**Pull request merged**